### PR TITLE
Tutorial March 6 Daily pull request

### PR DESCRIPTION
**Pull request by pe7r**
Added work in NonProject with props, added work with dynamic formation of component sets, added work with map() method in ToDoList app. Waiting for review.

### CODE REVIEW

**Comment by pe7r** [comment]
Dont merge it in the master, just review please

File: src/components/NonProject/NonProject.js [hunk 1]
```diff
@@ -1,17 +1,52 @@
...
-	return (
-	    <div className="joke">
-		        <h3 style={{display: props.question ? "block" : "none"}}>
-		         	Question: {props.question} 
-		        </h3>
-		        <h3 style={{color: !props.question && "blue"}}> Answer: {props.punchline} </h3>
-		    	<hr/>            
-        </div>
-	)
+// #1
```
Comment (ruslan-palahin) [inline src/components/NonProject/NonProject.js]: Please avoid comments. Try to write self-explainable and well readable code so in 98% of cases you do not need comments.

File: src/components/Product.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import React from 'react'
+
+function Product(props) {
+	console.log(props)
+	return (
+		<div className="todo-item">
```
Comment (ruslan-palahin) [inline src/components/Product.js]: In a real projects you should keep same indents through all files. Most projects have "eslint" tool which takes care about it automatically.

File: src/components/ProductList.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+import Product from './Product.js'
+import products from '../vschoolProducts.js'
+
+function ProductList() {
+
+	const productComponents = products.map(product => <Product key={product.id} name={product.name} 
+		price={product.price.toLocaleString("en-US", { style: 'currency', currency: 'USD' })} description={product.description} />)
+
+	return (	
+			<div>
```
Comment (ruslan-palahin) [inline src/components/ProductList.js]: Same about code style. Try select all code with ctra+a and ctrl+alt+l (small L) to auto-align code.

File: src/jokesData.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
+const jokesData = [
```
Comment (ruslan-palahin) [inline src/jokesData.js]: Looks like you do not need jokes any more. Please delete it if so.

File: src/vschoolProducts.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
+const products = [
```
Comment (ruslan-palahin) [inline src/vschoolProducts.js]: Same - please delete if not used.

**Review by ruslan-palahin — CHANGES_REQUESTED** [review]

File: src/components/NonProject/NonProject.js (hunk 1 above)
Comment (pe7r) [reply src/components/NonProject/NonProject.js]: I know. It was in the lesson code, so i left it. Sorry

Comment (pe7r) [reply src/components/NonProject/NonProject.js]: Fixed.

File: src/components/Product.js (hunk 2 above)
Comment (pe7r) [reply src/components/Product.js]: Will try next time(file was deleted)

File: src/components/ProductList.js (hunk 3 above)
Comment (pe7r) [reply src/components/ProductList.js]: Will try next time(file was deleted)

File: src/jokesData.js (hunk 4 above)
Comment (pe7r) [reply src/jokesData.js]: File deleted

File: src/vschoolProducts.js (hunk 5 above)
Comment (pe7r) [reply src/vschoolProducts.js]: File deleted

**Pull request merged**